clio-client/src/main/scala/org/broadinstitute/clio/client/util/FutureWithErrorMessage.scala

```diff
@@ -3,15 +3,18 @@ package org.broadinstitute.clio.client.util
 import com.typesafe.scalalogging.Logger
 
 import scala.concurrent.{ExecutionContext, Future}
+import scala.util.{Failure, Success}
 
-object FutureWithErrorMessage {
+trait FutureWithErrorMessage {
   implicit class FutureWithErrorMessage[A](future: Future[A]) {
     def withErrorMsg(message: String)(implicit ec: ExecutionContext,
                                       logger: Logger): Future[A] = {
-      future recover {
-        case ex =>
+      future andThen {
+        case Success(_) => ()
+        case Failure(ex) =>
+          if (Option(ex.getMessage).isDefined)
+            logger.error(ex.getMessage)
```

Comment:
This should handle logging out any error messages thrown during processing.

Reply:
I've modified the thrown exceptions to contain messages.